mozilla_aws_cli/listener.py

```diff
@@ -144,7 +160,45 @@ def handle_oidc_redirect_callback():
         })
 
     # callback into the login.callback() function in login.py
+    logger.debug("redirect_callback : request is {}".format(request.json))
     login.get_id_token(**request.json)
+    login.validate_id_token()
+    logger.debug("id_token_dict : {}".format(login.id_token_dict))
+    if login.id_token_dict is None:
+        logger.debug('Validation of token failed : {}'.format(login.token))
+        # TODO : What should we do in this case? How should the UI handle this?
+        return jsonify({
+            "result": "id_token_validation_failed",
+            "status_code": 400,
+        })
+
+    login.get_role_map()
+    try:
+        login.exchange_token_for_credentials()
+    except STSWarning as e:
+        if e.args[1] == 'ExpiredTokenException':
```

Comment:
I think all of this logic should probably be in `login.py` instead of `listener.py`? I generally like to separate things for clarity's sake.

Reply:
I figured I'd do the most 1:1 change first and then try consolidating these two sections of code that are somewhat duplicated between login.py and listener.py. My goal being to make it clear what was being chnanged in this PR which would be lost of if in the same PR I also consolidated the two sections into one.

Do you mind if I leave this here, file an issue to consolidate the login.py and this section, and complete that in a subsequent PR?

Reply:
Sure, I think that's probably fine.  :)

Reply:
Cool, filed #152